apps/backend/controllers/obsController.ts

```diff
@@ -0,0 +1,253 @@
+
+    // TODO: authentication and authorization
+
+    const { snapshotID, plantQuantity, plantID, soilType, dateBloomed, datePlanted } = req.body;
+
+    if (!snapshotID || !plantQuantity || !plantID){
+      res.status(400).json({ error: 'Required fields are missing'});
+      return;
+    }
+
+    const { data: obsID, error: obsError } = await supabase
+      .from('Observations')
+      .insert([
+        {
+          snapshotID,
+          plantQuantity,
+          plantID: plantID,
+          soilType: soilType || null,
+          dateBloomed: dateBloomed || null,
+          datePlanted: datePlanted || null,
+        },
+      ])
+      .select('observationID')
+      .single();
+
+    if (obsError) {
+      res.status(400).json({ error: obsError.message });
+      return;
+    }
+
+    res.status(201).json({
+      message: 'Observation created successfully',
+      obsID: obsID,
+    });
+
+  } catch (error) {
+    res.status(500).json({ error: `Internal server error: ${error}` })
+  }
+}
+
+export async function delObservation( req: AuthRequest, res: Response ){
+  try {
+
+    // TODO: authentication and authorization
+
+    const { obsID } = req.params;
+
+    if (!obsID) { // if obsID not passed
+      res.status(400).json({ error: 'Missing observation ID'});
+      return;
+    }
+
+    if (!/^\d+$/.test(obsID)){ // if obsID not an integer
+      res.status(400).json({ error: 'Observation ID must be an integer'});
+      return;
+    }
+
+    const { data: observation, error: obsError1 } = await supabase // find observation to make sure it exists
+      .from('Observations')
+      .select()
+      .eq('observationID', obsID)
+      .single();
+
+    if (obsError1) {
+      res.status(400).json({ error: obsError1.message });
+      return;
+    }
+    
+    if (!observation) { // observation does not exist
+      res.status(400).json({ error: `Observation ID: ${obsID} is not valid`})
+      return;
+    }
+
+    const currentTimestampz = new Date().toISOString(); // current time
+
+    const { error: obsError2 } = await supabase
+      .from('Observations')
+      .update({ deletedOn: currentTimestampz })
+      .eq('observationID', obsID);
+
+    if (obsError2) {
+      res.status(400).json({ error: obsError2.message });
+      return;
+    }
+
+    res.status(200).json({ 
+      message: "Observation deleted successfully",
+      deletedOn: currentTimestampz,
+    })
+
+  } catch (error) {
+    res.status(500).json({ error: `Internal server error: ${error}` })
+  }
+}
+
+export async function updateObservation( req: AuthRequest, res: Response ){
+  try {
+    // TODO: authentication and authorization
+
+    const { snapshotID, plantQuantity, plantID, soilType, dateBloomed, datePlanted } = req.body;
+
+    const { obsID } = req.params;
+
+    
+
+    if (!obsID) { // if obsID not passed
+      res.status(400).json({ error: 'Missing observation ID'});
+      return;
+    }
+
+    if (!/^\d+$/.test(obsID)){ // if obsID not an integer
```

Comment:
Since you're using this more than once, I would abstract this into a function in utils file